src/main/java/org/springframework/data/repository/config/CustomRepositoryImplementationDetector.java

```diff
@@ -80,23 +84,43 @@ public AbstractBeanDefinition detectCustomImplementation(RepositoryConfiguration
 		// TODO 2.0: Extract into dedicated interface for custom implementation lookup configuration.
 
 		return detectCustomImplementation(configuration.getImplementationClassName(), //
-				configuration.getBasePackages(), //
-				configuration.getExcludeFilters());
+				configuration.getImplementationBeanName(), configuration.getBasePackages(), //
+				configuration.getExcludeFilters(), configuration.getConfigurationSource().getBeanNameGenerator());
 	}
 
 	/**
 	 * Tries to detect a custom implementation for a repository bean by classpath scanning.
-	 * 
+	 *
+	 * @param nameGenerator
 	 * @param className must not be {@literal null}.
+	 * @param beanName
 	 * @param basePackages must not be {@literal null}.
 	 * @return the {@code AbstractBeanDefinition} of the custom implementation or {@literal null} if none found.
 	 */
-	public AbstractBeanDefinition detectCustomImplementation(String className, Iterable<String> basePackages,
-			Iterable<TypeFilter> excludeFilters) {
+	public AbstractBeanDefinition detectCustomImplementation(
+			String className,
+			String beanName,
+			Iterable<String> basePackages,
+			Iterable<TypeFilter> excludeFilters,
+			RepositoryBeanNameGenerator nameGenerator) {
 
 		Assert.notNull(className, "ClassName must not be null!");
 		Assert.notNull(basePackages, "BasePackages must not be null!");
 
+		Set<BeanDefinition> definitions = findCandidateBeanDefinitions(className, basePackages, excludeFilters);
+
+		SelectionSet<BeanDefinition> selection = new SelectionSet<>(definitions, c -> {
+			if (c.isEmpty()) {
+				return null;
+			}
+			return throwAmbiguousCustomImplementationException(c);
+		});
+		selection.filter(bd -> beanName != null && beanName.equals(nameGenerator.generateBeanName(bd)));
```

Comment:
Blank line before (generally alway after a block unless it's only a closing `}`).